### PR TITLE
Add slaught o matic

### PR DESCRIPTION
**Pull request by Uvash**
Add slaught o matic. A cheap self-defence weapon. Sprites - Infrared_baron
изображение

:cl:
add: Add slaught o matic. A cheap self-defence weapon. Now available in Frozen Star Guns&Ammo
:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/obj/item/weapon/gun/projectile/automatic/slaught_o_matic
+	name = ".35 slaught-O-matic"
+	desc = "A very cheap self-defence single use automatic pistol. Now less 90% metal. Uses .35 Auto rounds."
```
Comment (MadmanMartian) [inline code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm:3]:
```suggestion
	desc = "A very cheap self-defence single use automatic pistol. Now 90% less metal. Uses .35 Auto rounds."
```

File: code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	var/list/possible_colors = list("purple", "magenta", "blue", "cyan", "aqua", "green", "yellow", "orange", "red" )
+	var/choosen_color = ""
+
+	damage_multiplier = 0.8
+	penetration_multiplier = 0.2
+	recoil_buildup = 3
+	one_hand_penalty = 5 //despine it being handgun, it's better to hold in two hands while shooting. SMG level.
+
+	init_firemodes = list(
+		list(mode_name="full auto 300",  mode_type = /datum/firemode/automatic, fire_delay=2, icon="burst"),
```
Comment (MadmanMartian) [inline code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm:33]: Please use the defines.

File: code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		list(mode_name="full auto 800",  mode_type = /datum/firemode/automatic, fire_delay=0.8, icon="auto")
```
Comment (MadmanMartian) [inline code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm:34]: Please use the defines.

**Review by MadmanMartian — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm (hunk 1 above)
Comment (genessee596) [inline code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm:3]: Let's make this more lore-friendly.

```suggestion
	name = "FS HG .35 Auto \"Slaught-o-Matic\""
	desc = "This disposable plastic handgun is mass-produced by Frozen Star for civilian use. It often is used by street urchin, thugs, or terrorists on a budget. For what it's worth, it's not an awful handgun - but you only get one magazine before the gun locks up and becomes useless."
```

File: code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	..()
+
+	var/iconstring = initial(icon_state)
+	var/itemstring = "_" + choosen_color
+
+	icon_state = iconstring + itemstring
+	set_item_state(itemstring)
+
+/obj/item/weapon/gun/projectile/automatic/slaught_o_matic/load_ammo(obj/item/A, mob/user)
+	to_chat(user, SPAN_WARNING("You can't reload a disposable gun"))
```
Comment (genessee596) [inline code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm:56]:
```suggestion
	to_chat(user, SPAN_WARNING("You try to reload the handgun, but the magazine that's already loaded won't come out!"))
```

File: code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+/obj/item/weapon/gun/projectile/automatic/slaught_o_matic/unload_ammo(mob/user, var/allow_dump=1)
+	to_chat(user, SPAN_WARNING("You can't unload a disposable gun"))
```
Comment (genessee596) [inline code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm:59]:
```suggestion
	to_chat(user, SPAN_WARNING("You try to take out the handgun's magazine, but it won't budge!"))
```

**Review by genessee596 — CHANGES_REQUESTED** [review]
quality control

File: code/game/machinery/vending.dm [hunk 6]
```diff
@@ -1026,6 +1027,7 @@
 					/obj/item/weapon/tool/knife/tacknife = 5)
 
 	prices = list(
+					/obj/item/weapon/gun/projectile/automatic/slaught_o_matic = 199,
```
Comment (genessee596) [inline code/game/machinery/vending.dm:1030]: Why the weird price? Let's make it cost the same amount as it does in Cyberpunk - 90$. Nobody is going to pay 200$ for one magazine worth of gun.

```suggestion
					/obj/item/weapon/gun/projectile/automatic/slaught_o_matic = 90,
```

File: code/game/machinery/vending.dm [hunk 7]
```diff
@@ -1003,6 +1003,7 @@
 					/obj/item/weapon/gun/projectile/automatic/atreides = 3,
 					/obj/item/weapon/gun/projectile/shotgun/pump/gladstone = 3,
 					/obj/item/weapon/gun/projectile/shotgun/pump = 3,
+					/obj/item/weapon/gun/projectile/automatic/slaught_o_matic = 10,
```
Comment (genessee596) [inline code/game/machinery/vending.dm:1006]: Let's increase the amount of pistols in stock to drive home the fact that these things are a dime a dozen. Plus, there's only one FS vendor on the ship.

```suggestion
					/obj/item/weapon/gun/projectile/automatic/slaught_o_matic = 30,
```

File: code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm (hunk 3 above)
Comment (Uvash) [reply code/modules/projectiles/guns/projectile/automatic/slaught_o_matic.dm:34]: If we use FULL_AUTO_300 and FULL_AUTO_800 then the user will not be able to determine which mode is selected.

**Review by genessee596 — CHANGES_REQUESTED** [review]
Please add my two suggestions regarding weapon prices and weapon stock in the vendor.

**Review by MadmanMartian — APPROVED** [review]

**Pull request merged**